File: Aditya Vardhan Singh/Other Problems/Day 15/Delete all occurrences of a given key in a doubly linked list/C++/Delete all occurrences of a given key in a doubly linked list.cpp
// User function Template for C++


/* a Node of the doubly linked list 
struct Node
{
  int data;
  struct Node *next;
  struct Node *prev;
  Node(int x) { data = x; next = prev = NULL; }
}; */

class Solution {
  public:
    void deleteAllOccurOfX(struct Node** head_ref, int x) {
        struct Node * del;
        while ((*head_ref)->data == x) {
            del = *head_ref;
            *head_ref = (*head_ref)->next;
            del->next = del -> prev = nullptr;
            delete(del);
        }
        struct Node *temp = *head_ref;
        while (temp->next) {
            if (temp->next->data == x) {
                del = temp->next;
                temp->next = temp->next->next;
                del->next = del->prev = nullptr;
                delete(del);
            }
            else {
                temp = temp->next;
            }
        }
    }
};

File: Aditya Vardhan Singh/Other Problems/Day 15/Delete all occurrences of a given key in a doubly linked list/Python/Delete all occurrences of a given key in a doubly linked list.py
#User function Template for python3
'''
# Node Class
    class Node:
        def __init__(self, data):   # data -> value stored in node
            self.data = data
            self.next = None
            self.prev = None
'''
class Solution:
    #Function to delete all the occurances of a key from the linked list.
    def deleteAllOccurOfX(self, head, x):
        # code here
        while head.data == x:
            head = head.next
        temp = head
        while temp.next:
            if temp.next.data == key:
                temp.next = temp.next.next
            else:
                temp = temp.next
        return head
        # edit the linked list